include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -482,7 +483,7 @@ public :
             // Do not duplicate the closing point
             auto rot_end = boost::end(ring);
             std::size_t rot_index = index;
-            if (is_closed_in && size > 1)
+            if (BOOST_GEOMETRY_CONDITION(is_closed_in && size > 1))
```

Comment:
Related to `if constexpr`, isn't it, in cases like this, better to only apply the const part on the part being const?
I mean: `if (BOOST_GEOMETRY_CONDITION(is_closed_in) && size > 1)` ?

Reply:
done